dev/manual_tests/lib/density.dart

```diff
@@ -0,0 +1,572 @@
+
+  final String label;
+  final ValueChanged<bool> onChanged;
+  final bool value;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      children: <Widget>[
+        Checkbox(
+          onChanged: onChanged,
+          value: value,
+        ),
+        Text(label),
+      ],
+    );
+  }
+}
+
+class Options extends StatefulWidget {
+  const Options(this.model);
+
+  final OptionModel model;
+
+  @override
+  _OptionsState createState() => _OptionsState();
+}
+
+class _OptionsState extends State<Options> {
+  static final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+
+  @override
+  void initState() {
+    super.initState();
+    widget.model.addListener(_modelChanged);
+  }
+
+  @override
+  void didUpdateWidget(Options oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.model != oldWidget.model) {
+      oldWidget.model.removeListener(_modelChanged);
+      widget.model.addListener(_modelChanged);
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    widget.model.removeListener(_modelChanged);
+  }
+
+  void _modelChanged() {
+    setState(() {});
+  }
+
+  double sliderValue = 0.0;
+
+  String _densityToProfile(VisualDensity density) {
+    if (density == VisualDensity.standard) {
+      return 'standard';
+    }
+    if (density == VisualDensity.compact) {
+      return 'compact';
+    }
+    if (density == VisualDensity.comfortable) {
+      return 'comfortable';
+    }
+    return 'custom';
+  }
+
+  VisualDensity _profileToDensity(String profile) {
+    switch (profile) {
+      case 'standard':
+        return VisualDensity.standard;
+      case 'comfortable':
+        return VisualDensity.comfortable;
+      case 'compact':
+        return VisualDensity.compact;
+      case 'custom':
+      default:
+        return widget.model.density;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final SliderThemeData controlTheme = SliderTheme.of(context).copyWith(
+      thumbColor: Colors.grey[50],
+      activeTickMarkColor: Colors.deepPurple[200],
+      activeTrackColor: Colors.deepPurple[300],
+      inactiveTrackColor: Colors.grey[50],
+    );
+
+    return Padding(
+      padding: const EdgeInsets.fromLTRB(5.0, 0.0, 5.0, 10.0),
+      child: Builder(builder: (BuildContext context) {
+        return DefaultTextStyle(
+          style: TextStyle(color: Colors.grey[50]),
+          child: Column(
+            children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: Row(
+                  children: <Widget>[
+                    const Text('Text Scale'),
+                    Expanded(
+                      child: SliderTheme(
+                        data: controlTheme,
+                        child: Slider(
+                            label: '${widget.model.size}',
```

Comment:
for entire code block: 
```suggestion
                          label: '${widget.model.size}',
```